src/main/java/view/list/update/UserRatingUpdateListener.java

```diff
@@ -0,0 +1,77 @@
+package view.list.update;
+
+import javax.swing.event.TableModelEvent;
+import javax.swing.event.TableModelListener;
+
+import interface_adapter.list.ListTableModel;
+import view.ListView;
+
+/**
+ * A listener for rating column updates in the table.
+ */
+public class UserRatingUpdateListener implements TableModelListener {
+    private final ListView listView;
+
+    public UserRatingUpdateListener(ListView listView) {
+        this.listView = listView;
+    }
+
+    @Override
+    public void tableChanged(TableModelEvent e) {
+        if (listView.getMediaListTable().getModel() instanceof ListTableModel listTableModel) {
```

Comment:
Would this line ever evaluate to False? Just out of curiosity.

Reply:
Now that you mention it, probably not. I wrote the check because when the user logs out, the table model gets replaced with a default table model, and I don't think that'll be able to fire property change, huh.

Reply:
But I think I'll keep this here in case something wonky happens to the model in the future somehow. Maybe a RuntimeException should be thrown here in an else block.